pkg/secrets/fetch_secret.go

```diff
@@ -157,3 +191,41 @@ func fetchSecret(secretsHandle []string, origin string) (map[string]string, erro
 	}
 	return res, nil
 }
+
+func fileHashSHA256(f *os.File) (string, error) {
+	stat, err := f.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	h := sha256.New()
+
+	fileSize := stat.Size()
+	if fileSize == 0 {
+		return formatHash(h), nil
+	}
```

Comment:
Shouldn't this be an error? If the file size is 0, it's probably not a valid binary and we probably shouldn't return a valid hash.

Reply:
Can be done, though It's not an error from the context of the hash calculation.
```
touch /tmp/emptyfile
sha256sum /tmp/emptyfile
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855  /tmp/emptyfile
```

Reply:
I'm in favor of returning `hash("")` as it is now